### ISSUE TITLE
A StringField's value become "None" instead of None when  client submit it as null.

### ISSUE THREAD

**Issue by anlcnydn** [swistakm/graceful#57]
Let's assume we have a resource as implemented below:
 
```python
import falcon
from graceful.resources.generic import PaginatedListCreateAPI
from graceful.serializers import BaseSerializer
from graceful.fields import StringField


class TestSerializer(BaseSerializer):
    testStringField = StringField("Test String Field", source='test')


class TestList(PaginatedListCreateAPI):
    serializer = TestSerializer()

    def create(self, params, meta, validated, **kwargs):
        if validated.get('test') == "None":
            print("Is it a bug?")


app = application = falcon.API()

app.add_route("/v1/test", TestList())
```
 
When client sends a POST request with the following body:
```json
{
  "testStringField": null
}
```

Then the `test` field in the validated become `"None"` instead of `None`.

__Proposal__:

It can be a solution to check if data is `None` in the `from_representation`
method of the `StringField`. 

```python
def from_representation(self, data):
        """Convert representation value to ``str`` if it is not None."""
        return str(data) if data is not None else None
```

**Comment by swistakm** [comment]
@anlcnydn, thanks for this report.

This is something that definitely should be covered. Although, I'm not sure if the above approach to the problem is the right way. This is related to #3 issue (allow null/blank values in fields) that describes the same problem but with respect to all fields.

In my opinion the correct behaviour would be to not allow null values in any field if the field isn't explicitly marked to support such value by developer in the serializer class. Otherwise if we add this `if data is not None else None` in the `StringField.from_representation` we will have to do the same for every other field class. Also, this will not cover custom user fields and can lead to general inconsistencies in larger code bases.

What do you think about adding the `allow_null` argument to `BaseField` class initializer and handling everything inside of a serializer class?

Right now the only problem is that #3 was scheduled for 1.0.0 milestone that still requires some additional work. The ideal approach for `allow_null` argument would be to respond with `400 Bad Request` if the representation send by the client includes null value for non-nullable field. This will actually break the backwards compatibility. Still, we can introduce this new feature partially i.e. make fields with `allow_null=True` handle nulls properly as Nones and leave old behavior for `allow_null=False` until we switch to 1.x.

**Comment by swistakm** [comment]
Also, until we find the best approach to the problem and release it, you can use following shim in your code that will allow for proper null/None handling:
```python
class AllowNull(BaseField):
    """Wrapping pseudo-field that allows null values as a representation

    This is a workaround for lack of such suport in graceful but it is
    likely to change in future, see:

        https://github.com/swistakm/graceful/issues/3

    """
    def __init__(self, wraps):
        super(AllowNull, self).__init__(
            details=wraps.details,
            label=wraps.label,
            source=wraps.source,
            validators=None,
            many=wraps.many,
            read_only=wraps.read_only,
        )
        self.wraps = wraps
        self.type = wraps.type
        self.spec = wraps.spec

    def from_representation(self, data):
        if data is None:
            return None
        else:
            return self.wraps.from_representation(data)

    def to_representation(self, value):
        if value is None:
            return None
        else:
            return self.wraps.to_representation(value)

    def validate(self, value):
        return self.wraps.validate(value)

    def describe(self, **kwargs):
        return self.wraps.describe(**kwargs)
```

This is a field wrapper that can be used directly inside of your serializer class:

```python
class MySerializer(BaseSerializer):
    my_string_field = AllowNull(
        StringField("Test String Field", source='test')
    )
```

It's far from perfect but does the job.

**Comment by anlcnydn** [comment]
@swistakm, thanks for your quick answer.

I am easily convinced that my proposal was not the best choice to implement. 

When considering the options in [#3](https://github.com/swistakm/graceful/issues/3), marking fields as nullable and handling it in the serializer seems the best option in the ones which can be implemented.

Workaround that you have sent can solve the problem until the solution we discussed shows up. However, if I will be available, I can send a PR to give a hand for the work that will be done as well.

**Comment by swistakm** [comment]
Great. Any help is very welcome :)

**Comment by swistakm** [comment]
solved as part of #58
